Sources/Build/BuildPlan.swift

```diff
@@ -626,26 +626,57 @@ public final class SwiftTargetBuildDescription {
     public let isTestDiscoveryTarget: Bool
 
     /// True if this module needs to be parsed as a library based on the target type and the configuration
-    /// of the source code (for example because it has a single source file whose name isn't "main.swift").
-    /// This deactivates heuristics in the Swift compiler that treats single-file modules and source files
-    /// named "main.swift" specially w.r.t. whether they can have an entry point.
-    ///
-    /// See https://bugs.swift.org/browse/SR-14488 for discussion about improvements so that SwiftPM can
-    /// convey the intent to build an executable module to the compiler regardless of the number of files
-    /// in the module or their names.
+    /// of the source code
     var needsToBeParsedAsLibrary: Bool {
-        switch target.type {
+        switch self.target.type {
         case .library, .test:
             return true
         case .executable:
```

Comment:
Does this also need to apply to snippets BTW?  They're a lot like executables, AFAIK?

Reply:
I think so. Please see the last code example here: https://github.com/zkiraly/swift-argument-parser/blob/main/Sources/ArgumentParser/Documentation.docc/Articles/GettingStarted.md

With Xcode 14 Beta 4, if this example is implemented as a snippet, I get errors.

Reply:
None of this will apply to Xcode BTW, since we're not using the Build module there.